app/code/Magento/Cms/Model/Page/DataProvider.php

```diff
@@ -186,16 +203,19 @@ public function getMeta()
 
         //List of custom layout files available for current page.
         $options = [['label' => 'No update', 'value' => '_no_update_']];
-        if ($page = $this->findCurrentPage()) {
-            //We must have a specific page selected.
-            //If custom layout XML is set then displaying this special option.
+
+        try {
+            $page = $this->getCurrentPage();
             if ($page->getCustomLayoutUpdateXml() || $page->getLayoutUpdateXml()) {
                 $options[] = ['label' => 'Use existing layout update XML', 'value' => '_existing_'];
             }
             foreach ($this->customLayoutManager->fetchAvailableFiles($page) as $layoutFile) {
                 $options[] = ['label' => $layoutFile, 'value' => $layoutFile];
             }
+        } catch (LocalizedException $exception) {
+
```

Comment:
@gabrieldagama schould add logger there because empty catch will lead to static error ?

Reply:
Yes, I believe we should add a log in such a case.